Source/qol/xpbar.cpp

```diff
@@ -45,6 +44,27 @@ void DrawEndCap(const Surface &out, Point point, int idx, const ColorGradient &g
 	out.SetPixel({ point.x, point.y + 3 }, gradient[idx / 2]);
 }
 
+/**
+ * @brief Prints integer with thousands separator.
+ */
+std::string PrintWithSeparator(int n)
+{
+	std::string number = fmt::format("{:d}", n);
+	std::string out = "";
+
+	int length = number.length();
+	int mlength = length % 3;
+	if (mlength == 0)
+		mlength = 3;
+	out.append(number.substr(0, mlength));
+	for (int i = mlength; i < length; i += 3) {
+		out.append(_(/* TRANSLATORS: Thousands separator */ ","));
+		out.append(number.substr(i, 3));
+	}
+
+	return out;
+}
+

```

Comment:
Bit of an aside but this can be replaced with `fmt::format("{:L}", n)` if the global locale is configured properly (e.g. by adding `std::locale::global(std::locale(""));` into `InitXpBar()`).

Reply:
I looked at it and tried it, but that requires the locale be present on the system, and also that it is properly supported which isn't the case for a few systems (Windows doesn't properly support `locale`). And we support setting the language to `zh_TW` with out having system wide zh_TW, which results in a crash when using `"{:L}"`. So i begrudgingly ended up making my own implementation :)